### PR TITLE
Example nosimage.textproto and validator script.

### PR DESCRIPTION
**Pull request by wenovus**
**Limitation**: Validator script currently only validates OCPaths, and not RPCs.

Tested generator manually:
```
go run validate/validate.go -file example/example_nosimageprofile.textproto; rm -rf public
```
```
go run validate/validate.go -file example/example_nosimageprofile_invalid.textproto; rm -rf public
```

### CODE REVIEW

File: tools/internal/ocpaths/ocpaths.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+			for _, v := range oc.ΛEnum[enum] {
+				if v.Name == component {
+					ocpath.Key.Component = component
+					break CONSTRAINT_CHECK
+				}
+			}
+		}
+		fallthrough
+	default:
+		return nil, fmt.Errorf("path %q has invalid component %q", path, component)
```
Comment (dplore) [inline tools/internal/ocpaths/ocpaths.go:139]:
```suggestion
		return nil, fmt.Errorf("path %q has invalid component %q.  Want %q", path, component, component_regex)
```

Comment (dplore) [reply tools/internal/ocpaths/ocpaths.go:139]: to aid in debugging, include the required regex in the error log

File: tools/internal/ocpaths/ocpaths.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	if featureprofileID != "" {
+		matched, err := regexp.MatchString("^([a-z0-9]+_)*[a-z0-9]+$", featureprofileID)
+		if err != nil {
+			return nil, err
+		}
+		switch {
+		case matched:
+			ocpath.FeatureprofileID = featureprofileID
+		default:
+			return nil, fmt.Errorf("unexpected featureprofileID string %q for path %v", featureprofileID, path)
```
Comment (dplore) [inline tools/internal/ocpaths/ocpaths.go:153]: also log the required regex for featureprofileID

File: tools/internal/ocpaths/ocpaths.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+					break CONSTRAINT_CHECK
+				}
+			}
+		}
+		fallthrough
+	default:
+		return nil, fmt.Errorf("path %q has invalid component %q", path, component)
+	}
+
+	// Basic validation of featureprofileid if it exists
```
Comment (dplore) [inline tools/internal/ocpaths/ocpaths.go:142]: Be even more explicit.  featureprofileID is optional.  Only validate the string format if it exists.

File: tools/nosimage/validate/validate.go
Comment (dplore) [inline tools/nosimage/validate/validate.go:1]: Should we also validate ocrpc?

Comment (wenovus) [reply tools/nosimage/validate/validate.go:1]: Yes, I can add that as a follow-up.

File: tools/internal/ocpaths/ocpaths.go (hunk 1 above)
Comment (wenovus) [reply tools/internal/ocpaths/ocpaths.go]: Done.

File: tools/internal/ocpaths/ocpaths.go (hunk 2 above)
Comment (wenovus) [reply tools/internal/ocpaths/ocpaths.go]: Done.

File: tools/internal/ocpaths/ocpaths.go (hunk 3 above)
Comment (wenovus) [reply tools/internal/ocpaths/ocpaths.go]: Done.

**Review by dplore — APPROVED** [review]

**Pull request merged**